lib/ae_mdw/txs.ex

```diff
@@ -37,15 +39,27 @@ defmodule AeMdw.Txs do
 
   @create_tx_types ~w(contract_create_tx channel_create_tx oracle_register_tx name_claim_tx)a
 
-  @spec fetch_txs(direction(), query(), cursor() | nil, limit()) ::
+  @spec fetch_txs(direction(), scope(), query(), cursor() | nil, limit()) ::
           {:ok, [tx()], cursor() | nil} | {:error, reason()}
-  def fetch_txs(direction, query, cursor, limit) do
+  def fetch_txs(direction, scope, query, cursor, limit) do
     ids = query |> Map.get(:ids, MapSet.new()) |> MapSet.to_list()
     types = query |> Map.get(:types, MapSet.new()) |> MapSet.to_list()
     cursor = deserialize_cursor(cursor)
 
+    txis_range =
+      case scope do
+        {:gen, %Range{first: first_gen, last: last_gen}} ->
+          {gen_to_txi(first_gen, direction), gen_to_txi(last_gen, opposite_dir(direction))}
```

Comment:
I was a little confused by the direction change to the opposite one. Maybe it would be better to have explicit functions:
`first_gen_to_txi` and `last_gen_to_txi` as the direction is used to select the field.

Reply:
Not sure what you meant by this, I think I've done what you asked for, it doesn't really change things much. For next time, please provide code samples, it's a lot less ambiguous than Enligsh! :sunglasses: